### PR TITLE
Update navigation samples on Navigation overview page

### LINKED ISSUE #10057: Add excerpts for snippets on 'Navigation and routing' page
### Page URL

https://docs.flutter.dev/ui/navigation/

### Page source

https://github.com/flutter/website/tree/main/src/ui/navigation/index.md

### Describe the problem

The code snippets on this page don't have excerpts and therefore can easily get out of date or fail analysis in the first place.

### Expected fix

Add [code excerpts](https://github.com/dart-lang/site-shared/blob/main/doc/code-excerpts.md) to this page.

### PR DESCRIPTION
**Pull request by antfitch**
_Description of what this PR is changing or adding, and why:_

Update code snippets in the [navigation overview](https://docs.flutter.dev/ui/navigation) so that they run.

Notes:

* All examples follow the same format to make it easier for a developer to see the small difference between each routing strategy.
* Small textual updates in some areas to make information more clear and to follow the style guides.
* Lowered heading for Limitations so that it doesn't appear in the right TOC (it seems unnecessary for it to appear there).
* I am not sure how to create a variable for a link on the same page. Take a look at how I handled the 'Limitations' link.

_Issues fixed by this PR (if any):_

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/content/ui/navigation/index.md [hunk 1]
```diff
@@ -89,12 +197,71 @@ To use the Router, switch to the `router` constructor on `MaterialApp` or
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Second Screen'),
+      ),
+      body: Center(
+        child: ElevatedButton(
+          child: const Text('Go to first screen'),
```
Comment (antfitch) [inline src/content/ui/navigation/index.md:258]: Should be 'Open first screen'

**Review by sfshaza2 — APPROVED** [review]
lgtm. Did you merely re-order the code, or did you update the code? If the latter, can you have the code reviewed? Otherwise, lgtm

**Comment by antfitch** [comment]
> lgtm. Did you merely re-order the code, or did you update the code? If the latter, can you have the code reviewed? Otherwise, lgtm

There is new code in there. I'm not able to update the PR at this point (it's read-only for me) and I need to update two samples to match the first. To make things easier, I'll create a new PR.

**Comment by antfitch** [comment]
Obsolete. Trying again with a new PR (this one has become read-only for me)

**Pull request closed without merging**